src/oscar/apps/checkout/applicator.py

```diff
@@ -0,0 +1,53 @@
+from decimal import Decimal as D
+
+from oscar.core import prices
+from oscar.core.loading import get_class
+
+PaymentCharge = get_class("checkout.surcharges", "PaymentCharge")
+
+
+class Surcharges(object):
+    total = None
+    methods = {}
+
+    def __init__(self, methods, total):
+        self.total = total
+        self.methods = methods
+
+
+class SurchargeApplicator(object):
+
+    methods = (
+        PaymentCharge(charge_excl_tax=D("10.0"), charge_incl_tax=D("10.0"), charge_is_percentage=True),
```

Comment:
please just use excl_tax and incl_tax instead of charge_excl_tax so it behaves more like a price object